### PR TITLE
Suppress write mode warning

### LINKED ISSUE #1751: Add a way to suppress the rustfmt write-mode warning
I assume this warning will stick around for a while. Could you add a way to suppress it?

I started working on a patch to add it to rustfmt.toml, but I realized that this actually is a per-user setting, not a per-repository setting :(

### PR DESCRIPTION
**Pull request by topecongiro**
Closes #1751.

### CODE REVIEW

File: src/bin/rustfmt.rs [hunk 1]
```diff
@@ -120,6 +120,8 @@ fn make_opts() -> Options {
     opts.optflag("h", "help", "show this message");
     opts.optflag("V", "version", "show version information");
     opts.optflag("v", "verbose", "print verbose output");
+    // Supress warning. Remove this option after the default write mode changed to overwrite.
+    opts.optflag("", "no-warn", "inhibit warning about write-mode change");
```
Comment (nrc) [inline src/bin/rustfmt.rs]: could you make the name a bit more specific please? Perhaps "no-warn-write-mode", but given the verbosity, we could allow `-w` too.

Comment (topecongiro) [inline src/bin/rustfmt.rs]: Thanks for your comment! I will change and rebase shortly.

**Comment by topecongiro** [comment]
Rebased to fix test failure.

**Comment by nrc** [comment]
Thanks!

**Pull request merged**